### PR TITLE
Add KDL language extension

### PR DESCRIPTION
**Pull request by elkowar**
Adds language support for the KDL configuration language.

### CODE REVIEW

File: extensions.toml [hunk 1]
```diff
@@ -536,6 +536,10 @@ version = "0.1.1"
 submodule = "extensions/kanagawa-themes"
 version = "0.0.1"
 
+[kdl-syntax]
```
Comment (maxdeviant) [inline extensions.toml:539]: Any reason to not call this just `kdl` instead of `kdl-syntax`?

Comment (elkowar) [reply extensions.toml:539]: I thought there might at some point be a more fully featured kdl extension with some LSP or something, but I guess for now just calling it kdl works fine. I'll rename it!

Comment (elkowar) [reply extensions.toml]: Done

**Review by maxdeviant — APPROVED** [review]
Thanks!

**Pull request merged**